src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Configuration/ConfigurationProjectConfigurationDimensionProvider.cs

```diff
@@ -76,7 +75,9 @@ public override Task OnDimensionValueChangedAsync(ProjectConfigurationDimensionV
         /// <returns>A task for the async operation.</returns>
         private async Task OnConfigurationAddedAsync(UnconfiguredProject project, string configurationName)
         {
-            string evaluatedPropertyValue = await GetPropertyValue(project);
+            string? evaluatedPropertyValue = await GetPropertyValue(project);
+            if (evaluatedPropertyValue == null)
+                throw new InvalidOperationException($"Property {PropertyName} not defined.");
```

Comment:
Swap `NullReferenceException` with `InvalidOperationException`, here and in a few other places.

Reply:
👍 	for the better exception message